tests/test-org-ql.el

```diff
@@ -1686,19 +1686,19 @@ with keyword arg NOW in PLIST."
                 (unquoted-lambda-in-list-link "[[org-ql-search:todo:?buffers-files%3D%28%28lambda%20nil%20%28error%20%22UNSAFE%22%29%29%29]]"))
           (it "Errors for a quoted lambda"
             (expect (open-link quoted-lambda-link)
-                    :to-throw 'error '("CAUTION: Link not opened because unsafe buffers-files parameter detected: (lambda nil (error UNSAFE))")))
+                    :to-throw 'error '("Value lambda is not a valid buffer/file")))
           (it "Errors for an unquoted lambda"
             (expect (open-link unquoted-lambda-link)
-                    :to-throw 'error '("CAUTION: Link not opened because unsafe buffers-files parameter detected: (lambda nil (error UNSAFE))")))
+                    :to-throw 'error '("Value lambda is not a valid buffer/file")))
           (it "Errors for a quoted lambda in a list"
             (if (version< (org-version) "9.3")
                 (expect (open-link quoted-lambda-in-list-link)
-                        :to-throw 'error '("CAUTION: Link not opened because unsafe buffers-files parameter detected: ((quote (lambda nil (error UNSAFE))))"))
+                        :to-throw 'error '("Value (quote (lambda nil (error UNSAFE))) is not a valid buffer/file"))
               (expect (open-link quoted-lambda-in-list-link)
-                      :to-throw 'error '("CAUTION: Link not opened because unsafe buffers-files parameter detected: ('(lambda nil (error UNSAFE)))"))))
+                      :to-throw 'error '("Value (quote (lambda nil (error UNSAFE))) is not a valid buffer/file"))))
           (it "Errors for an unquoted lambda in a list"
             (expect (open-link unquoted-lambda-in-list-link)
-                    :to-throw 'error '("CAUTION: Link not opened because unsafe buffers-files parameter detected: ((lambda nil (error UNSAFE)))"))))
+                    :to-throw 'error '("Value (lambda nil (error UNSAFE)) is not a valid buffer/file"))))
```

Comment:
Why did you change these errors? These are important safety tests.

Reply:
I had added a explicit error when the input to the `org-ql-view--expand-buffers-files` function, 
https://github.com/alphapapa/org-ql/blob/564d491604f76aa05e6e0a8e9d87b762153a78fd/org-ql-view.el#L1108

According to my understanding, the previous errors being tested were because of the `pcase` failing. So roll back the error I triggering and the `pacse` to fail which would lead to the older errors?
Even then, in this implementation the return value is always a list, which makes the error being produced to be a bit different than what it previously had.

Reply:
I'm not sure what you mean.  Do you understand the potential security issue here?

Reply:
To clarify, the potential values for the `org-ql-view--expand-buffers-files` now are the following:
- A buffer
- A string
- A file-name
- one of "all", "org-agenda-files", "org-directory", "buffer", or an empty string (which evaluates to the same values as "buffer")
- a list containing above values

I am also converting all buffer objects to string values to avoid duplicates that are introduced because of the buffer/buffer-name of a file and a file-name being passed. 

As a result this function now always returns a list of strings (or at least should, will add tests for these :D).

This is in part due to moving to using `completing-read-multiple` which returns a list by default. Because of this, I had dropped the `listp` predicate in the `pcase` inside the function since now all elements in the list are being processed individually. This in-turn results in the unsafe values for buffers/files being tested in the link safety section never getting to the point where that error gets signalled since they never match any of the values in the current pcase. Which was why I was signalling an error when an invalid value is passed. Does this make sense?

Reply:
I may have overlooked that functions are a valid input to `buffers-files` in `org-ql-select`. Which might be creating a whole bunch of new problems :)

Currently (in the master branch) the following should create a view with the function
```elisp
(org-ql-search #'a-function-returning-list-of-files '(level 1))
```
with the value for `org-ql-view-buffers-files` is set to `#'a-function-returning-list-of-files`. 

If I understand this correctly, this is left as such so refreshing the view would recall the function? (or alternatively the evaluated value can be stored?)

I am not sure how a function would translate to a value in `completing-read-mulitple`, so I am skipping prompting for that entirely when refreshing a view with a function as a value for `org-ql-view-buffers-files`.

The safety concern you were referring to before comes from allowing a function to be used as a parameter for `org-ql-select`? Currently the `org-ql-view--expand-buffers-files` function would signal an error if used with a function, and when storing a link, the function's evaled value is used for `buffers-files`.

Does that cover the problem with unsafe values for `buffers-files`?

Reply:
You can see where the error is signaled here: https://github.com/alphapapa/org-ql/blob/94f9e6f3031b32cf5e2149beca7074807235dcb0/org-ql-view.el#L638

That safety check should be left there, in that function, where links are followed.  IIUC, the change you made to the tests would cause that check to not be tested, which could lead to security vulnerabilities in the future.

Reply:
That check looks at either the value returned by the `org-ql-view--expand-buffers-files` or the current-buffer. If I understand it correctly, it was there to make sure the value returned by the `org-ql-view--expand-buffers-files` function is not another function that can get evaluated once passed to `org-ql-select`. Currently, if I understand when and how `org-ql-view--expand-buffers-files` gets called, in a valid situation, it only should get values that I had previously mentioned, anything else would result in the error on the new tests being signaled.

If the concern is future changes to the `org-ql-view--expand-buffers-files` function results in returning a function value, that function can be mocked to make sure the error in https://github.com/alphapapa/org-ql/blob/94f9e6f3031b32cf5e2149beca7074807235dcb0/org-ql-view.el#L638
get's signaled?